src/platform/cmx/chal_cpu.h

```diff
@@ -0,0 +1,39 @@
+#ifndef CHAL_CPU_H
+#define CHAL_CPU_H
+
+#include <pgtbl.h>
+#include <thd.h>
+
+
+u32_t comp1_stack[1000];
+
+/* FIXME: I doubt these flags are really the same as the PGTBL_* macros */
+static inline u32_t
+chal_cpu_fault_errcode(struct pt_regs *r) { return r->orig_r4; }
+
+static inline u32_t
+chal_cpu_fault_ip(struct pt_regs *r) { return r->r15_pc; }
+
+static inline void
+chal_user_upcall(void *ip, u16_t tid)
+{
+	/* Now we switch the execution to user space, and begin to use the PSP */
+	__asm__ __volatile__( \
+					    "ldr r0,=comp1_stack \n\t" \
```

Comment:
User-level finds its own stack when we activate it.  That will require fixing up the `components/lib/` stack code.  It is reasonable to do this in the next phase after this PR, but it is something we can discuss next time we meet.